Lib/posixpath.py

```diff
@@ -196,16 +187,16 @@ def isjunction(path):
     return False
 
 
-# Being true for dangling symbolic links is also useful.
+def isreserved(path):
+    """Return true if the pathname is reserved by the system.
+    Always returns False on posix"""
+    return False
```

Comment:
This was deliberately excluded from the `posixpath` interface - see https://github.com/python/cpython/pull/95486#issuecomment-1341553054

Reply:
This alternative implementation I was using might be interesting:

```python
def isreserved(path: StrOrBytesPath) -> bool:
    r"""Test whether a path is reserved.

    macOS:
    - Under a file: "foo.txt/bar" -> NotADirectoryError -> True
    - Too long file name: "a" * 256 -> OSError -> True
    - Too long path: "a/" * 512 -> OSError -> True
    - Too many levels of symlinks: "s/" * 33 -> OSError -> True

    Windows:
    - Too many levels of symlinks: "s\\" * 64 -> OSError -> True
    - Incorrect name: "\foo" -> OSError -> True
    - Incorrect name: "foo*bar" -> OSError -> True
    - Name ending with period: "foo." -> False (WRONG!)
    - Name ending with space: "foo " -> False (WRONG!)
    - File stream: "foo:bar" -> False (WRONG!)
    - Character device: "nul" -> False (WRONG?)
    - DOS device name: "aux" -> False
    - DOS device name with extension: "nul.txt" -> False
    """
    try:
        os.stat(path)
    except FileNotFoundError:
        return False
    except (OSError, ValueError):
        return True

    return False
````

It also detects some issues on posix, sadly it doesn't detect all the issues on windows as the previous implementation.
Therefore I'll remove the posix implementation again.